pkg/registry/core/pod/storage/eviction.go

```diff
@@ -232,18 +239,41 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 		return rtStatus, nil
 	}
 
-	// At this point there was either no PDB or we succeeded in decrementing
+	// At this point there was either no PDB or we succeeded in decrementing or
+	// the pod was unready and we have enough healthy replicas
+
+	deletionOptions := originalDeleteOptions.DeepCopy()
```

Comment:
I would move this copy operation into the if block and rename it to `deleteOptions`.

```
deleteOptions := originalDeleteOptions

if updateDeleteOptions {
  deleteOptions = deleteOptions.DeepCopy()
  setPreconditionsResourceVersion(...)
}
```

No need to do a deep copy except when you are explicitly changing the object.